include/bout/field.hxx

```diff
@@ -527,22 +527,36 @@ T pow(BoutReal lhs, const T& rhs, const std::string& rgn = "RGN_ALL") {
  * and uses checkData() to, if CHECK >= 3, check
  * result for non-finite numbers
  *
+ * If the input field has parallel slices, then those will also be
+ * operated on. To avoid this, use `withoutParallelSlices()` to
+ * discard slices before calling.
+ *
  */
 #ifdef FIELD_FUNC
 #error This macro has already been defined
 #else
-#define FIELD_FUNC(_name, func)                                        \
-  template <typename T, typename = bout::utils::EnableIfField<T>>      \
-  inline T _name(const T& f, const std::string& rgn = "RGN_ALL") {     \
-    AUTO_TRACE();                                                      \
-    /* Check if the input is allocated */                              \
-    checkData(f);                                                      \
-    /* Define and allocate the output result */                        \
-    T result{emptyFrom(f)};                                            \
-    BOUT_FOR(d, result.getRegion(rgn)) { result[d] = func(f[d]); }     \
-    result.name = std::string(#_name "(") + f.name + std::string(")"); \
-    checkData(result);                                                 \
-    return result;                                                     \
+#define FIELD_FUNC(_name, func)                                                          \
+  template <typename T, typename = bout::utils::EnableIfField<T>>                        \
+  inline T _name(const T& f, const std::string& rgn = "RGN_ALL") {                       \
+    AUTO_TRACE();                                                                        \
+    /* Check if the input is allocated */                                                \
+    checkData(f);                                                                        \
+    /* Define and allocate the output result */                                          \
+    T result{emptyFrom(f)};                                                              \
+    BOUT_FOR(d, result.getRegion(rgn)) { result[d] = func(f[d]); }                       \
+    if (f.hasParallelSlices()) {                                                         \
+      /* Operate on parallel slices */                                                   \
+      result.splitParallelSlicesAndAllocate();                                           \
+      for (size_t i{0}; i != f.numberParallelSlices(); ++i) {                            \
+        BOUT_FOR(d, result.getRegion(rgn)) { result.yup(i)[d] = func(f.yup(i)[d]); }     \
```

Comment:
but here we need a shifted region (at least for FCI) ...

Reply:
My thinking with this was that if we are operating over a given region of a field, then we also want to operate over all the yup/down fields connected to those points. Hence the region indices should be the same for yup/down. I guess you have a different model for how this should work?